students/Ben_freeman/lesson04/mailroom2.py

```diff
@@ -0,0 +1,88 @@
+#Donor dict
+donor_list={
+"Ahmed":[123,456,789],
+"Sally": [10,11],
+"Bob": [12],
+"Vladamir": [13.14,1516.17],
+"Santa": [181920.21,22,23.24],
+}
+
+
+#send thanks funct
+def send_thanks():
+    while True:
+        query=input("Please enter a Full Name, enter 'list' to see a list of current donors, or quit to exit: ")
+        if query in donor_list:
+            new_donate=float(input("How much would you like to donate: \n"))
+            donor_list[query].append(new_donate)
+            print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                "Sincearly,\n\n"
+                "A low paid intern\n")
+            break
+        elif query=="list":
+            print('\n'.join(donor_list))
+        elif query.lower()=="quit":
+            return
+        elif query not in donor_list:
+            donor_list[query]=[]
+            new_donate=float(input("\nHow much would you like to donate: \n"))
+            donor_list[query].append(new_donate)
+            print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                "Sincearly,\n\n"
+                "A low paid intern\n")
+            break
```

Comment:
you can re-organize this a bit to reduce some duplicated code:
```
while True:
    query = input('...')
    if query == "list":
        # print donors
    elif query == "quit":
        return
    else:
        donation = input('...')
        if donor in donor_list:
            # update donation
        else:
            # new donor
```

Reply:
Nice, I'll get on that!